### PR TITLE
Update code compliance to use TestResult and update Modify method compliance check

### LINKED ISSUE #326: Update code compliance to use TestResult from Test_oM
Following changes in Test_oM, update CodeCompliance_oM and Engine to match.

### LINKED ISSUE #327: Update Modify method compliance check
In line with [this issue](https://github.com/BHoM/admin/issues/11).

### PR DESCRIPTION
**Pull request by FraserGreenroyd**
Fixes #326 
Fixes #327
Fixes https://github.com/BHoM/admin/issues/14

New Wiki page [here](https://github.com/BHoM/documentation/wiki/ModifyReturnsDifferentType).

It will take time for BHoMBot to be upgraded in line with the changes to this toolkit by @IsakNaslundBh and myself in this PR and #321 so the original wiki page for modify methods will remain up until BHoMBot is running the check in this PR instead. This is to avoid BHoMBot providing a link which doesn't exist in the interim. The old wiki page will come down once BHoMBot is updated against these PRs.

Here's some test assets to see this refactor in action: [fun* link](https://burohappold.sharepoint.com/sites/BHoM/02_Current/Forms/AllItems.aspx?newTargetListUrl=%2Fsites%2FBHoM%2F02%5FCurrent&viewpath=%2Fsites%2FBHoM%2F02%5FCurrent%2FForms%2FAllItems%2Easpx&viewid=a5a93cba%2Dfcca%2D46cc%2Da31a%2D80b8d5cbfd7b&id=%2Fsites%2FBHoM%2F02%5FCurrent%2F12%5FScripts%2F02%5FPull%20Request%2FBHoM%2FTest%5FToolkit%2F%23328%2DCodeComplianceRefactor)

*fun not guaranteed

### CODE REVIEW

File: CodeComplianceTest_Engine/CodeComplianceTest_Engine.csproj [hunk 1]
```diff
@@ -143,6 +143,11 @@
     <Reference Include="System.Data" />
     <Reference Include="System.Net.Http" />
     <Reference Include="System.Xml" />
+    <Reference Include="Test_oM, Version=4.0.0.0, Culture=neutral, processorArchitecture=MSIL">
```
Comment (FraserGreenroyd) [inline CodeComplianceTest_Engine/CodeComplianceTest_Engine.csproj:146]:
```suggestion
    <Reference Include="Test_oM">
```

File: CodeComplianceTest_Test/CodeComplianceTest_Test.csproj [hunk 2]
```diff
@@ -103,6 +103,10 @@
     <Reference Include="System.Threading.Tasks.Extensions, Version=4.2.0.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
       <HintPath>..\packages\System.Threading.Tasks.Extensions.4.5.3\lib\netstandard2.0\System.Threading.Tasks.Extensions.dll</HintPath>
     </Reference>
+    <Reference Include="Test_oM, Version=4.0.0.0, Culture=neutral, processorArchitecture=MSIL">
```
Comment (FraserGreenroyd) [inline CodeComplianceTest_Test/CodeComplianceTest_Test.csproj:106]:
```suggestion
    <Reference Include="Test_oM">
```

File: CodeCompliance_oM/CodeComplianceTest_oM.csproj [hunk 3]
```diff
@@ -60,6 +60,11 @@
     <Reference Include="System.Data" />
     <Reference Include="System.Net.Http" />
     <Reference Include="System.Xml" />
+    <Reference Include="Test_oM, Version=4.0.0.0, Culture=neutral, processorArchitecture=MSIL">
```
Comment (FraserGreenroyd) [inline CodeCompliance_oM/CodeComplianceTest_oM.csproj:63]:
```suggestion
    <Reference Include="Test_oM">
```

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a CI/CD instruction. I am authorising dispensation to be granted on check ref. 1791487487

**Comment by FraserGreenroyd** [comment]
@BHoMBot fix copyright headers ref. 1791487805

**Comment by FraserGreenroyd** [comment]
@BHoMBot fix copyright headers ref. 1791487805

**Comment by FraserGreenroyd** [comment]
@BHoMBot fix copyright headers ref. 1791487805

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a CI/CD instruction. I am authorising dispensation to be granted on check ref. 1806630558

**Review by rolyhudson — APPROVED** [review]
All seems to be working as expected. 
Also noting that the description provided in the wiki page is clear in communicating the outcomes and reasons for the ModifyReturnsDifferentType check.

**Comment by FraserGreenroyd** [comment]
Thanks @rolyhudson :smile:

**Pull request merged**